pipenv/environments.py

```diff
@@ -55,9 +53,6 @@
 PIPENV_SKIP_VALIDATION = True
 # Tells Pipenv where to load .env from.
 PIPENV_DOTENV_LOCATION = os.environ.get('PIPENV_DOTENV_LOCATION')
-# Use shell compatibility mode when using venv in project mode.
-if PIPENV_VENV_IN_PROJECT:
-    PIPENV_SHELL_COMPAT = True
```

Comment:
Not sure why this was here in the first place (need to ask @kennethreitz), but some additional checks are likely required in `do_shell` if you delete this.